File: server/core/src/main/java/dev/slimevr/posestreamer/ServerPoseStreamer.java
package dev.slimevr.posestreamer;

import dev.slimevr.VRServer;
import dev.slimevr.util.ann.VRServerThread;
import dev.slimevr.tracking.processor.skeleton.HumanSkeleton;


public class ServerPoseStreamer extends TickPoseStreamer {

	protected final VRServer server;

	public ServerPoseStreamer(VRServer server) {
		super(null); // Skeleton is registered later
		this.server = server;

		// Register callbacks/events
		server.addSkeletonUpdatedCallback(this::onSkeletonUpdated);
		server.addOnTick(this::onTick);
	}

	@VRServerThread
	public void onSkeletonUpdated(HumanSkeleton skeleton) {
		this.skeleton = skeleton;
	}

	@VRServerThread
	public void onTick() {
		super.doTick();
	}
}

File: server/core/src/main/java/dev/slimevr/posestreamer/ServerPoseStreamer.kt
package dev.slimevr.posestreamer

import dev.slimevr.VRServer
import dev.slimevr.tracking.processor.skeleton.HumanSkeleton
import dev.slimevr.util.ann.VRServerThread

class ServerPoseStreamer(val server: VRServer) : TickPoseStreamer(server.humanPoseManager.skeleton) {

	init {
		// Register callbacks/events
		server.addSkeletonUpdatedCallback { skeleton: HumanSkeleton? ->
			this.onSkeletonUpdated(
				skeleton,
			)
		}
		server.addOnTick { this.tick() }
	}

	@VRServerThread
	fun onSkeletonUpdated(skeleton: HumanSkeleton?) {
		if (skeleton != null) {
			this.skeleton = skeleton
		}
	}

	@VRServerThread
	fun tick() {
		super.tick(server.fpsTimer.timePerFrame)
	}
}
